js/runner.go

```diff
@@ -418,12 +418,12 @@ func (u *VU) RunOnce(ctx context.Context) error {
 	}
 
 	// Call the default function.
-	_, _, err := u.runFn(ctx, u.Runner.defaultGroup, u.Default, u.setupData)
+	_, _, err := u.runFn(ctx, u.Runner.defaultGroup, true, u.Default, u.setupData)
 	return err
 }
 
 func (u *VU) runFn(
-	ctx context.Context, group *lib.Group, fn goja.Callable, args ...goja.Value,
+	ctx context.Context, group *lib.Group, isDefault bool, fn goja.Callable, args ...goja.Value,
```

Comment:
I really dislike the signature change here and in `GetTrail`, but I didn't see an easier way to determine whether it's running setup/teardown (in which case iterations should be emitted) or the default function (in which case it should be emitted).

Reply:
Yeah... This seems fine for now, but we'd probably need to refactor the JS runner long-term (after or in #1007) - this method signature is getting quite hefty, and it doesn't make a lot of sense to emit `iterations` or `iteration_duration` in a `netext.NetTrail` :smile: . I think the `netext.Dialer` should just have a method that returns a struct with the `DataSent` and `DataReceived`, and something else should emit the composite metric sample container with them and the iteration metrics. Anyway, for now, can you add a `TODO: refactor this` here, or rather in the `Dialer.GetTrail()` method signature :sweat_smile: ?

Reply:
I added a TODO in eb8aa668, but just linking to here, since I'm not sure I could paraphrase that properly. Feel free to fix it by adding a commit to this branch.

Reply:
good enough as a TODO, these links usually work fine